src/main/java/ticTacToe/guiApp/ActionSetter.java

```diff
@@ -0,0 +1,21 @@
+package ticTacToe.guiApp;
+
+import ticTacToe.game.Mark;
+import ticTacToe.grid.Grid;
+
+public class ActionSetter {
+
+    private final Grid grid;
+
+    public ActionSetter(Grid grid) {
+        this.grid = grid;
+    }
+
+    public void addClickHandler(ButtonWrapper button, String buttonNumber) {
+        button.setOnAction(event -> {
+            Mark currentMark = grid.markThatMoves();
+            grid.addMark(currentMark, buttonNumber);
+            button.actualButton().setText(currentMark.sign);
```

Comment:
Could you also call your new `GuiGameFlow` class in here? And then the class can check if the game is over (with the help of `Grid`).